inginious/frontend/plugins/grader_generator/pages/graderforms.py

```diff
@@ -0,0 +1,131 @@
+import json
+from collections import OrderedDict
+from abc import ABC, abstractmethod
+from inginious.frontend.pages.course_admin.task_edit import CourseEditTask
+
+# TODO: Replace this class and all json dumps to main class only one time
+
+class InvalidGraderError(Exception):
+    def __init__(self, message, *args):
+        super().__init__(message, *args)
+
+        self.message = message
+
+
+class GraderForm(ABC):
+    """ This class manage the form for the creation of the grader """
+
+    def __init__(self, task_data, task_fs):
+        self.task_data = dict(task_data)
+        self.task_fs = task_fs
+
+    @abstractmethod
+    def parse(self):
+        pass
+
+    @abstractmethod
+    def validate(self):
+        pass
+
+
+class MultilangForm(GraderForm):
```

Comment:
What if we separate this file in more file, where each file contains only one class. This is in case we add more classes in one file, this will be unreadable.

Reply:
I'm planning on doing that but in this moment i will just add complexity in list of files, so I decide against it